internal/runnerv2service/execution.go

```diff
@@ -184,6 +184,10 @@ func (e *execution) Wait(ctx context.Context, sender runnerv2.RunnerService_Exec
 	e.closeIO()
 
 	if waitErr != nil {
+		// Drain the readSendDone channel to avoid goroutine leaks.
```

Comment:
Found when running e2e tests thanks to the logger reporting calls after the test case finished.